### PR TITLE
Reactive Messaging codestart

### LINKED ISSUE #20872: Reactive Messaging codestart (in java and kotlin)
### Description

Create a codestart for Reactive Messaging in java and kotlin

### Implementation ideas

It's better to first discuss the content of the codestart here.

The goal is to have code:

- that is re-used 90% of the time to get started with the extension (nearly no deletion, renaming is fine)
- that is compatible with all other compatible extensions codestarts

What is the piece of code that you nearly always start with when using the extension.. Make it compile, and it's the perfect codestart!

The guide to create an extension codestart quarkus.io/guides/extension-codestart

### PR DESCRIPTION
**Pull request by ia3andy**
* fix #20872  
* Generate `MyReactiveMessagingApplication` to demonstrates `@Emitter`, `@Incoming` and `@Outgoing` annotations.

### CODE REVIEW

File: extensions/reactive-messaging-http/runtime/src/main/resources/META-INF/quarkus-extension.yaml [hunk 1]
```diff
@@ -10,4 +10,9 @@ metadata:
     - "web"
     - "reactive"
     - "messaging"
-  status: "experimental"
\ No newline at end of file
+  status: "experimental"
```
Comment (loicmathieu) [inline extensions/reactive-messaging-http/runtime/src/main/resources/META-INF/quarkus-extension.yaml:13]: There is no provided codestart for HTTP so why adding it here ?

File: integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/ReactiveMessagingCodestartIT.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    @Test
+    void testKafkaContent() throws Throwable {
+        kafkaCodestartTest.checkGeneratedSource("org.acme.MyReactiveMessagingApplication");
+        kafkaCodestartTest.assertThatGeneratedFileMatchSnapshot(JAVA, "src/main/resources/application.properties");
+        kafkaCodestartTest.checkGeneratedTestSource("org.acme.MyReactiveMessagingApplicationTest");
+    }
+
+    @Test
+    void testMQTTContent() throws Throwable {
+        mqttCodestartTest.assertThatGeneratedFileMatchSnapshot(JAVA, "src/main/resources/application.properties");
```
Comment (loicmathieu) [inline integration-tests/devtools/src/test/java/io/quarkus/devtools/codestarts/quarkus/ReactiveMessagingCodestartIT.java:42]: Why not asserting the `MyReactiveMessagingApplication` and ` MyReactiveMessagingApplicationTest` in MQQT and AMQP ?
They should also be generated so tested.

**Review by loicmathieu — APPROVED** [review]
LGTM, I added some comment but it can go as is.

File: extensions/reactive-messaging-http/runtime/src/main/resources/META-INF/quarkus-extension.yaml (hunk 1 above)
Comment (ia3andy) [reply extensions/reactive-messaging-http/runtime/src/main/resources/META-INF/quarkus-extension.yaml:13]: Good point!

**Comment by ia3andy** [comment]
> LGTM, I added some comment but it can go as is.

fixed :)

**Pull request merged**